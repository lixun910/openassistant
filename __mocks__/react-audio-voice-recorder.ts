export const useAudioRecorder = jest.fn();
