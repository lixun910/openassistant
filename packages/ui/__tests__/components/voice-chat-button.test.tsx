/**
 * @jest-environment jsdom
 */
import React, { act } from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
jest.mock('react-audio-voice-recorder');

import VoiceChatButton from '../../src/components/voice-chat-button';
import { useAudioRecorder } from 'react-audio-voice-recorder';

(useAudioRecorder as jest.Mock).mockReturnValue({
  isRecording: false,
  recordingBlob: new Blob(['test'], { type: 'audio/wav' }),
  startRecording: jest.fn(),
  stopRecording: jest.fn(() => new Blob(['test'], { type: 'audio/wav' })),
});

describe('VoiceChatButton', () => {
  it('renders correctly', () => {
    // override the default mock for useAudioRecorder
    const { container } = render(
      <VoiceChatButton onRecordingComplete={() => Promise.resolve()} />
    );
    expect(container).toMatchSnapshot();
  });

  it('starts recording on button click', async () => {
    const { startRecording } = useAudioRecorder();
    await act(async () => {
      render(<VoiceChatButton onRecordingComplete={jest.fn()} />);
    });

    await act(async () => {
      fireEvent.click(screen.getByText('Talk to Ask'));
    });
    expect(startRecording).toHaveBeenCalled();
  });

  it('stops recording on button click when already recording', async () => {
    const { stopRecording } = useAudioRecorder();
    render(<VoiceChatButton onRecordingComplete={jest.fn()} />);

    await act(async () => {
      fireEvent.click(screen.getByText('Talk to Ask'));
    });
    await act(async () => {
      fireEvent.click(screen.getByText('Talk to Ask'));
    });
    expect(stopRecording).toHaveBeenCalled();
  });

  it('calls onRecordingComplete with the recording blob', async () => {
    const mockOnRecordingComplete = jest.fn();
    render(<VoiceChatButton onRecordingComplete={mockOnRecordingComplete} />);

    await act(async () => {
      fireEvent.click(screen.getByText('Talk to Ask')); // Start recording
    });
    await act(async () => {
      fireEvent.click(screen.getByText('Talk to Ask')); // Stop recording
    });

    expect(mockOnRecordingComplete).toHaveBeenCalledWith(expect.any(Blob));
  });
});
