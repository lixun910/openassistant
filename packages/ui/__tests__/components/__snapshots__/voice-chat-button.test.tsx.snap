// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoiceChatButton renders correctly 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default/40 text-default-700 data-[hover=true]:opacity-hover"
    data-testid="voice-chat-button"
    type="button"
  >
    <span />
    Talk to Ask
  </button>
</div>
`;
